### ISSUE TITLE
Wrong predictions when using BatchNormalization with training flag set 

### ISSUE THREAD

**Issue by zaidalyafeai** [tensorflow/tfjs#562]
To get help from the community, check out our [Google group](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs).

#### TensorFlow.js version
latest 
#### Browser version
Version 66.0.3359.139 
#### Describe the problem or feature request
Batchnorm has wrong predictions when setting training = 1 
#### Code to reproduce the bug / link to feature request
I created this simple keras model 
```
def SimpleModel():
  x = Input(shape = (2, 2, 3))
  y = BatchNormalization()(x, training = 1)
  y = Flatten()(y)
  z = Dense(units = 1)(y)
  return Model(inputs = x, outputs = z)
```
After training, the batch norm layer weights are 
```
[array([0.99683774, 0.99683774, 0.99683774], dtype=float32),
 array([-0.00316227,  0.00316227, -0.00316228], dtype=float32),
 array([-0.08008331,  0.01483306,  0.12279604], dtype=float32),
 array([1.0677528, 1.0555032, 0.9067482], dtype=float32)]
```
After running the prediction 
`model.predict(np.zeros((1, 2, 2, 3)))`
The output 
```
array([[[[-0.00316227,  0.00316227, -0.00316228],
         [-0.00316227,  0.00316227, -0.00316228]],

        [[-0.00316227,  0.00316227, -0.00316228],
         [-0.00316227,  0.00316227, -0.00316228]]]], dtype=float32)
```
On the browser the weights are the same but the activations are 
```
Tensor
    [[[[0.0740574, -0.0112231, -0.1316395],
       [0.0740574, -0.0112231, -0.1316395]],

      [[0.0740574, -0.0112231, -0.1316395],
       [0.0740574, -0.0112231, -0.1316395]]]]
```

Explanation 

on keras when setting training = 1, it uses the statics of the prediction sample 

![image](https://user-images.githubusercontent.com/15667714/43457962-0f87e6c8-94d1-11e8-8a17-0277ac62289c.png)

Tensorflow.js uses the stored moving mean and variance of the training data 

![image](https://user-images.githubusercontent.com/15667714/43458007-3e0db7ac-94d1-11e8-94fe-427bd1d609d9.png)

**Comment by zaidalyafeai** [comment]
The same problem happens in TensorFlow as well.

**Comment by caisq** [comment]
@zaidalyafeai you mean tf.keras?

**Comment by zaidalyafeai** [comment]
No, this definition 

https://www.tensorflow.org/api_docs/python/tf/layers/batch_normalization

**Comment by zaidalyafeai** [comment]
@caisq, I am trying to understand the source code. Could you please explain to me what is broadcasting ?

**Comment by nsthorat** [comment]
@caisq did we ever resolve this issue? I assume tf.layers is doing the right thing in TensorFlow..

**Comment by zaidalyafeai** [comment]
I resolved this issue by modifying the source code and changing the definition of batch norm during inference time. My pix2pix demo is based on that!

**Comment by caisq** [comment]
Training with BatchNormazliation should be working. See the ACGAN example under review at
https://github.com/tensorflow/tfjs-examples/pull/187

I'd like to see the code you're using and the change you made in order for it to work, @zaidalyafeai , if possible.

**Comment by zaidalyafeai** [comment]
@caisq, I may have accidentally deleted the source code :/  but the idea is simple I just forced batch norm layer to use the statistics of the input sample as if it was training. So, I didn't add any code just re-routing.

**Comment by rthadur** [comment]
Closing this due to lack of activity, feel to reopen. Thank you

**Comment by FinderBMap** [comment]
Small value of BatchNormalization parameter  
"momentum"   =  0.01
may help this (works for 1D in TensorFlow js)
